sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -39,6 +39,14 @@ object Klaviyo {
      * @param applicationContext
      */
     fun initialize(apiKey: String, applicationContext: Context) = safeCall {
+        if (Registry.isRegistered<Config>() && Registry.get<Config>().apiKey != apiKey) {
+            Registry.get<State>().let { state ->
```

Comment:
This logic could live in side effects coordinator now, right?

Reply:
ooh sorry right because unregister needs the old api key value. hm. 
It would be a pretty easy add to "broadcast" the old value of the property. I had that originally, but none of my callbacks were using it, so I simplified.

Reply:
i can work to rewriting the request to take in the api key in the constructor instead?

Reply:
Ah yeah good point. LMK If you want to pair/discuss adding in the broadcast of the property's old value as well.

Reply:
I remember now that I didn't do it partly because none of my callbacks needed the old value and partly because I didn't want to deal with trying to make `StateObserver` generic. Maybe it'll just  have to be `typealias StateObserver = (key: Keyword?, oldValue: Any?) -> Unit`

Reply:
I actually still had this partially implemented in a stash locally. I cleaned it up and put it in a [draft PR](https://github.com/klaviyo/klaviyo-android-sdk/pull/164) to share